### PR TITLE
Rj add pundit

### PR DESCRIPTION
**Pull request by jrzricardo**

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -4,7 +4,7 @@
   </div>
 </div>
 
-<% if current_user == @user || !@user.private? || current_user.leaders.include?(@user) %>
+<% if policy(@user).show_photos? %>
```
Comment (jniewiel) [inline app/views/users/show.html.erb:7]: Good job creating a policy to hide private users' feeds!

File: app/policies/user_policy.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    true
+  end
+  
+  def feed?
+    true
+  end
+  
+  def liked?
+    true
+  end
```
Comment (jniewiel) [inline app/policies/user_policy.rb:19]: In this lesson, we're not implementing Likes, so this isn't needed.

File: app/policies/photo_policy.rb
Comment (jniewiel) [inline app/policies/photo_policy.rb:1]: Nice job with your photo policies, Ricardo!

**Review by jniewiel — CHANGES_REQUESTED** [review]
Hello Ricardo! Overall, very good job with your Pundit pull request. Comments and posts are good. However, there are a few issues that popped up with follows:

Current users are shown a "follow" button on their own profiles.

There is an issue where the current user can't accept or deny their follow requests. (No authorization. Hint: check your policies)

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,10 +1,20 @@
 class ApplicationController < ActionController::Base
+
+  include Pundit::Authorization
```
Comment (ajmompr) [inline app/controllers/application_controller.rb:3]: good

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 class CommentsController < ApplicationController
   before_action :set_comment, only: %i[ show edit update destroy ]
-  before_action :is_an_authorized_user, only: %i[ destroy create]
-  before_action :ensure_current_user_is_author, only: %i[ destroy update edit ]
+
+  before_action { authorize @comment || Comment }
```
Comment (ajmompr) [inline app/controllers/comments_controller.rb:4]: good

File: app/controllers/follow_requests_controller.rb [hunk 5]
```diff
@@ -1,6 +1,8 @@
 class FollowRequestsController < ApplicationController
   before_action :set_follow_request, only: %i[ show edit update destroy ]
 
+  before_action { authorize @follow_request || FollowRequest}
```
Comment (ajmompr) [inline app/controllers/follow_requests_controller.rb:4]: good

File: app/controllers/photos_controller.rb [hunk 6]
```diff
@@ -1,6 +1,7 @@
 class PhotosController < ApplicationController
   before_action :set_photo, only: %i[ show edit update destroy ]
   before_action :ensure_current_user_is_owner, only: %i[destroy update edit]
+  before_action { authorize @photo || Photo }
```
Comment (ajmompr) [inline app/controllers/photos_controller.rb:4]: good

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -1,5 +1,6 @@
 class UsersController < ApplicationController
   before_action :set_user, only: %i[ show liked feed followers following discover ]
+  before_action { authorize @user || User }
```
Comment (ajmompr) [inline app/controllers/users_controller.rb:3]: good

File: app/policies/comment_policy.rb [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    @user = user
+    @comment = comment
+  end
+  
+  def create?
+    true
+  end
+ 
+  def destroy?
+    user == comment.author
```
Comment (ajmompr) [inline app/policies/comment_policy.rb:14]: good

File: app/policies/user_policy.rb (hunk 2 above)
Comment (ajmompr) [reply app/policies/user_policy.rb:19]: +1

File: app/views/users/show.html.erb (hunk 1 above)
Comment (ajmompr) [reply app/views/users/show.html.erb:7]: +1

**Review by ajmompr — APPROVED** [review]
Great job

File: app/views/follow_requests/_form.html.erb [hunk 9]
```diff
@@ -12,8 +12,9 @@
   <% end %>
 
   <%= form.hidden_field :recipient_id %>
-
+  <% unless @user == current_user %>
```
Comment (jniewiel) [inline app/views/follow_requests/_form.html.erb:15]: Nice job fixing the self follow issue!

File: app/policies/follow_request_policy.rb
Comment (jniewiel) [inline app/policies/follow_request_policy.rb:1]: All of the previous follow request issues are now working properly. Great job!

**Review by jniewiel — APPROVED** [review]
Great job, Ricardo!

**Pull request merged**